src/kernel/src/sched/sched.c

```diff
@@ -0,0 +1,12 @@
+#include <sched.h>
+
+ExecutionState * save_context(void * sp, void * lr, void * fp) {
+    // I'm for now assuming this includes r10
+    asm volatile ("stm r0, {r4 - r11}^");
```

Comment:
same remark as before. Usually it's a bad idea to make these low level things functions as c is free to change registers whenever it's more efficient to do so. Function calls are good opportunity for this. Since you will probably only need to save the execution state in one place I'd say you can just inline it.

Reply:
so this is an old function that we're not using anymore, I'm not sure if I already comitted it. But we're using an assembly subroutine